modules/product/src/Form/AddToCartForm.php

```diff
@@ -258,22 +258,26 @@ protected function selectVariationFromUserInput(array $variations, FormStateInte
     $user_input = $form_state->getUserInput();
     $current_variation = reset($variations);
     if (!empty($user_input)) {
+      // We cascade through the attributes and find the variations through
+      // attribute dependency. That means the last possible attribute helps
+      // define the actual variation, so we do not use it as a qualifying
+      // attribute, because it defines our final available variations.
       $attributes = $user_input['attributes'];
-      foreach ($variations as $variation) {
-        $match = TRUE;
-        foreach ($attributes as $field_name => $value) {
+      array_pop($attributes);
+
+      // Loop the through attributes to find qualifying variations.
+      foreach ($attributes as $field_name => $value) {
+        foreach ($variations as $key => &$variation) {
```

Comment:
Shoot, no. For some reason I has have a derp moment trying to unset the array item :P